docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,214 @@
+
+    kernel-cmdline:
+        append:
+            - nohz=on
+            - nohz_full=2-N
+            - irqaffinity=0-1
+
+
+Refer to :doc:`../reference/kernel-boot-parameters` for the list of supported parameters.
+
+Modify ``snapcraft.yaml`` to fit your application.
+At least, make sure to change the name and version to something distinct, for example to ``realtime-pc`` and ``example`` respectively.
+
+
+Now, build the gadget snap:
+
+.. code-block:: console
+
+    $ snapcraft --verbose
+    ...
+    Created snap package realtime-pc_example_amd64.snap
+
+
+.. tip::
+    You need to rebuild the snap every time you change the `snapcraft.yaml` file.
+
+
+Create the model assertion
+--------------------------
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Create the model assertion in the same directory as the pc-gadget.
+
+Set the following:
+
+
+- ``authority-id``, ``brand-id`` to your developer ID
+- ``timestamp`` to a RFC3339 formatted string, withing you signing key's validity
+- ``store`` to you IoT App Store ID
+
+Change the value of ``model`` to something representative of your model.
+
+The ``snaps`` array is a list of snaps that get included in the image.
+The gadget snap has no ``channel`` and ``id``, because it isn't in the store.
+We're have built it locally and will pass it to the image builder.
+In practice, the gadget snap should be uploaded to the store and then listed in the model assertion along with its channel and id.
+Uploading to the store makes it possible to use a signed snap that receives updates.
+
+The ``realtime-kernel`` snap contains the realtime Linux kernel.
+This snap should be included explicitly in your IoT App Store.
+
+Add any other snaps, including for example your realtime application.
+
+Next, we need to sign the model assertion.
+Refer to the guide on `signing model assertion`_ for details on how to sign the model assertion. 
+
+Here are the needed steps:
+
+1) Create and register a key
+
+
+.. code-block:: shell
+
+    snapcraft create-key realtime-ubuntu
+    snapcraft register-key realtime-ubuntu
+
+
+You can use ``snapcraft list-keys`` to check your existing keys.
+
+2) Sign the model assertion
+
+.. code-block:: shell
+
+    snap sign -k realtime-ubuntu model.json > model.signed.yaml
+
+The ``snap sign`` command takes JSON as input and produces YAML as output!
+
+.. tip::
+
+    You need to repeat the signing every time you change the input model, because the signature is calculated based on the model.
+
+Before we continue, let's have an overview of our files :
+
+.. code-block:: console
+
+    $ tree -L 1
+    .
+    ├── model.json
+    ├── model.signed.yaml
+    └── pc-gadget
+
+    2 directories, 2 files
```

Comment:
This command output seems a little bit misleading to me, because it doesn't reflect the contents of the cloned `pc-gadget`  directory.

Reply:
It is not meant to. The tree is set to level 1.

Reply:
Yes, but even at the level 1 you get the other files that belong to the pc-gadget repository:

```console
$ tree -L 1
.
├── gadget
├── icon.png
├── legacy-bios
├── README.md
├── recovery.md
├── snakeoil
├── snap
└── snapcraft.yaml
```

Reply:
In the documented case, `tree -L 1` is executed from `pc-gadget`'s parent directory, i.e. the working directory of this how-to. I guess this is the same confusion pointed out in https://github.com/canonical/real-time-ubuntu-docs/pull/51#discussion_r1705555682

Reply:
See https://github.com/canonical/real-time-ubuntu-docs/pull/51/commits/e21e4a8211ffc2a97466430508cbfa23730ef625